packages/task/src/browser/task-service.ts

```diff
@@ -409,6 +410,46 @@ export class TaskService implements TaskConfigurationClient {
     }
 
     async runTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+
+        // check if the task is active
+        const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+            const taskConfig = taskInfo.config;
+            return this.taskDefinitionRegistry.compareTasks(taskConfig, task);
+        });
+        if (matchedRunningTaskInfo) { // the task is active
+            const taskName = this.taskNameResolver.resolve(task);
+            const taskId = matchedRunningTaskInfo.taskId;
+            const terminalId = matchedRunningTaskInfo.terminalId;
+            let terminal: TerminalWidget | undefined;
+            if (terminalId) {
+                terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                if (terminal) {
+                    terminal.show(); // bring the terminal to the top
+                }
+            }
+            this.messageService.info(`The task \'${taskName}\' is already active`, 'Terminate Task', 'Restart Task').then(async actionTask => {
+                if (actionTask) {
+                    await this.kill(taskId);
+                    if (terminal) {
+                        const parent = terminal.parent;
```

Comment:
That looks fragile, terminals can be moved between areas, some product cat integrate terminal in another widget.

We should not assume anything about areas or structure of widgets here.

Reply:
> That looks fragile, terminals can be moved between areas, some product cat integrate terminal in another widget.

Yes that's why I am checking `parent.id === BOTTOM_AREA_ID`. In my opinion, it makes sense to hide the parent only if it is in the bottom panel. What do you think?